src/test/regress/expected/multi_colocation_utils.out

```diff
@@ -343,3 +343,227 @@ SELECT find_shard_interval_index(1300016);
+    ORDER BY colocationid;
+ colocationid | shardcount | replicationfactor | distributioncolumntype 
+--------------+------------+-------------------+------------------------
+            1 |          2 |                 2 |                     23
+            2 |          2 |                 1 |                     23
+            3 |          2 |                 2 |                     25
+            4 |          4 |                 2 |                     23
+(4 rows)
+
+SELECT logicalrelid, colocationid FROM pg_dist_partition
+    WHERE colocationid >= 1 AND colocationid < 1000 
+    ORDER BY colocationid;
+ logicalrelid  | colocationid 
+---------------+--------------
+ table1_groupa |            1
+ table2_groupa |            1
+ table1_groupb |            2
+ table2_groupb |            2
+ table1_groupc |            3
+ table2_groupc |            3
+ table1_groupd |            4
+ table2_groupd |            4
+ table3_groupd |            4
+(9 rows)
+
+-- check effects of dropping tables
+DROP TABLE table1_groupA;
+SELECT * FROM pg_dist_colocation WHERE colocationid = 1;
+ colocationid | shardcount | replicationfactor | distributioncolumntype 
+--------------+------------+-------------------+------------------------
+            1 |          2 |                 2 |                     23
+(1 row)
+
+-- dropping all tables in a colocation group also deletes the colocation group
+DROP TABLE table2_groupA;
+SELECT * FROM pg_dist_colocation WHERE colocationid = 1;
+ colocationid | shardcount | replicationfactor | distributioncolumntype 
+--------------+------------+-------------------+------------------------
+(0 rows)
+
+-- create dropped colocation group again
+SET citus.shard_count = 2;
+CREATE TABLE table1_groupE ( id int );
+SELECT create_distributed_table('table1_groupE', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+CREATE TABLE table2_groupE ( id int );
+SELECT create_distributed_table('table2_groupE', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+-- check metadata
+SELECT * FROM pg_dist_colocation 
+    WHERE colocationid >= 1 AND colocationid < 1000 
+    ORDER BY colocationid;
+ colocationid | shardcount | replicationfactor | distributioncolumntype 
+--------------+------------+-------------------+------------------------
+            2 |          2 |                 1 |                     23
+            3 |          2 |                 2 |                     25
+            4 |          4 |                 2 |                     23
+            5 |          2 |                 2 |                     23
+(4 rows)
+
+-- check created shards
+SELECT
+    logicalrelid,
+    pg_dist_shard.shardid AS shardid,
+    shardstorage,
+    nodeport,
+    shardminvalue,
+    shardmaxvalue
+FROM
+    pg_dist_shard,
+    pg_dist_shard_placement
+WHERE
+    pg_dist_shard.shardid = pg_dist_shard_placement.shardid AND
+    pg_dist_shard.shardid >= 1300026
+ORDER BY
+    shardid,
```

Comment:
Very minor: Hmm, if we ORDER BY `logicalrelid, shardmaxvalue`, reading the output might be easier.